content/posts/2024-11-05-flang-new.md

```diff
@@ -0,0 +1,446 @@
+is today) in use. Classic Flang's binary name was `flang`, just like the folder F18 now occupied
+in the LLVM monorepo.
+
+* **October 2020** Flang is included in an LLVM release for the first time in
+  LLVM 11.0.0. There was an `f18` binary and the aforementioned script
+  `flang.sh`.
+* **October 2021** First appearance of the `flang-new` binary, in the LLVM 13.0.0
+  release.
+* **April 2022** NVIDIA [ceases development](https://discourse.llvm.org/t/nvidia-transition-from-fir-dev/61947)
+  of the `fir-dev` branch in the Classic Flang project. Upstreaming of `fir-dev`
+  to the LLVM Project begins around this date.
+* **April 2022** `flang-new` can now do code generation if an experimental flag
+  is used. Enabled by the code upstreamed into the LLVM Project from `fir-dev`.
+* **May 2022** The `flang.sh` driver script becomes `flang-to-external-fc`. It
+  allows the user to use `flang-new` to parse then unparse Fortran source code then send
+  the result to an existing Fortran compiler such as GFortran.
+
+  (this seems strange, but it allowed users to drop in Flang as a replacement
+  for an existing compiler and test Flang's parsing code on their own codebases)
+* **June 2022** `flang-new` was changed to do code generation
+  [by default](https://github.com/llvm/llvm-project/commit/0867d4157328169c570f4d1e9c01806624035873)
+  with no extra flags required.
+* **July 2022** Valentin Clement (NVIDIA) [announces](https://discourse.llvm.org/t/nvidia-transition-from-fir-dev/61947/5)
+  that upstreaming of `fir-dev` is complete.
+
+At this point we had in the LLVM Project the Flang you see today. The
+combination of the F18 frontend and MLIR based code generation from `fir-dev`.
+As opposed to Classic Flang with its PGI based frontend and legacy backend.
+
+An initiative to upstream Classic Flang had effectively rewritten the entire
+compiler by the time it was complete.
+
+<!-- For some reason Hugo doesn't render any plaintext in these bullet points
+     correctly unless I they have blank lines between them. -->
+
+* **April 2024** The `flang-to-external-fc` script is removed. Marking the point
+  where `flang-new` was mature enough to be used as a standalone compiler.
+
+* **October 2024** The community deems that Flang has met the criteria to not be
+  "new" and the name is changed. Goodbye `flang-new`, hello `flang`!
+
+* TODO: First release with `flang` binary goes here!
+
+# Flang and the Definition of New
+
+The renaming of Flang was a contentious
+[proposal](https://discourse.llvm.org/t/proposal-rename-flang-new-to-flang/69462)
+and an early use of the
+[LLVM proposal process](https://github.com/llvm/llvm-www/blob/main/proposals/LP0001-LLVMDecisionMaking.md).
+
+The most critial point was that `flang` has meant different things to different
+people over time.
+
+To some `flang` was the out of tree Classic Flang. To some it was the in-tree
+LLVM Flang. Some were aware of both, but really wanted to move on from Classic
+Flang.
```

Comment:
IMHO, it would be good to introduce "LLVM Flang" (with quotes) somewhere in the text, similarly to how "Classic Flang" is introduced.

Main rationale - once published, people will likely use this blog post as their main reference for what "Flang" actually is. So this might be the only opportunity to clarify the distinction and to avoid confusion.

As for "Flang" being 10 yrs old ... I see where you're coming from, makes sense 👍🏻

Reply:
Yes this is a good benchmark. Would this article be a good (albeit long) answer to "which flang is/was which".

Reply:
I've improved the place where I first use the term.